### PR TITLE
Add "id" field and logging to DLC-BTC-POR BTC RPC request

### PR DESCRIPTION
**Pull request by cawthorne**
"id" is a required field in the jsonrpc spec. We should include it in the request to support all RPC specs.

https://www.jsonrpc.org/specification#notification

### CODE REVIEW

File: packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts [hunk 1]
```diff
@@ -210,7 +210,12 @@ export class DLCBTCPorTransport extends SubscriptionTransport<TransportTypes> {
       txId,
       requestConfig,
     )
-    return response.data.result
+
+    const result = response.data.result
+
+    if (!result) console.log('BITCOIN_RPC_URL - getrawtransaction failed: ', response)
```
Comment (karen-stepanyan) [inline packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts:216]: We need to use a `logger` instance

Comment (cawthorne) [reply packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts]: Done! went with the logger.error method, but open to a different method.

**Comment by mxiao-cll** [comment]
Need a changeset

**Comment by mmcallister-cll** [comment]
would prob be best if it was just 1 log line, but OK as is i guess

File: packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts [hunk 2]
```diff
@@ -210,7 +210,15 @@ export class DLCBTCPorTransport extends SubscriptionTransport<TransportTypes> {
       txId,
       requestConfig,
     )
-    return response.data.result
+
+    const result = response.data.result
+
+    if (!result) {
+      logger.error('BITCOIN_RPC_URL - getrawtransaction failed, requestConfig: ', requestConfig)
```
Comment (karen-stepanyan) [inline packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts:217]: We don't need the `requestConfig` but maybe only `txId`

**Comment by cawthorne** [comment]
> Need a changeset

I intentionally didn't add a changeset, as I was not currently planning on merging this patch, until it was conceptually approved.

File: packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts (hunk 2 above)
Comment (cawthorne) [reply packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts:217]: In a current exchange with a NOP on this, printing out the `requestConfig` might have highlighted the situation faster. However, it isn't a huge value add either.

Comment (karen-stepanyan) [reply packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts]: It looks like they have issues with their RPC. Do you still think this change will add a value ?

Comment (cawthorne) [reply packages/sources/dlc-btc-por/src/transport/proof-of-reserves.ts]: I do think it adds, value as their RPC actually follows the jsonrpc spec perfectly, and it wasn't an uptime issue. So it is technically an issue with the EA not with them. The changes they are making now to accommodate the missing id field in their RPC is a work around. Even though most RPCs can handle a missing id field, we should probably support all RPCs which technically follow the spec.

**Review by karen-stepanyan — APPROVED** [review]

**Pull request merged**